libs/barista-components/table/src/table.ts

```diff
@@ -321,4 +332,41 @@ export class DtTable<T> extends _DtTableBase<T> implements OnDestroy {
 
   /** CSS class added to any row or cell that has sticky positioning applied. */
   protected stickyCssClass = 'dt-table-sticky';
+
+  exportCSV(): void {
+    let ds = this.dataSource as DtTableDataSource<T>;
+    let exportData = ds.filteredData;
+    let csv = '';
+    let keys: string[] = [];
+    //get list of keys
+    if (typeof exportData[0] == 'object') {
+      keys = Object.keys(exportData[0]);
+    }
+    if (!keys.length) return;
+    csv += keys.join(',') + '\n';
+
+    exportData.forEach((row) => {
+      let vals = Object.values(row);
+      keys.forEach((key, idx) => {
+        let val = row[key];
+        if (typeof val != 'string') val = val.toString();
+        if (val.includes(',')) csv += `"${val}"`;
+        else csv += val;
+        if (idx < vals.length) csv += ',';
+      });
+      csv += '\n';
+    });
+
+    //make csv document
+    let o = new Blob([csv], { type: 'text/csv' });
```

Comment:
We might want to split this into a utility function. It does not necessarily be part of the member.